src/compiler/crystal/progress_tracker.cr

```diff
@@ -18,9 +18,7 @@ module Crystal
       print_stats
       print_progress
 
-      start_time = Time.now
-      retval = yield
-      time_taken = Time.now - start_time
+      time_taken = Time.measure { retval = yield }
```

Comment:
This shouldn't work because `retval` is not known outside the block. I guess you need to use `Time.monotonic` here. Might work with defining `retval` as `uninitialized`.

Reply:
This might work::
```
retval = uninitialized typeof(yield)
time_taken = measure do
  retval = yield
end
```

Reply:
Don't use `uninitialized` willy-nilly. `retval = nil`

Reply:
`retval.nil` would add `Nil` to the return types of the method and then you'd have to use `retval.not_nil!`
So it's probably best to just use `Time.monotonic` without a block

Reply:
fixed.